x/curating/keeper/upvote.go

```diff
@@ -44,14 +44,14 @@ func (k Keeper) CreateUpvote(
 		if err != nil {
 			return err
 		}
+		// shadow deposit as its no longer available
+		deposit = deposit.Sub(deposit)
```

Comment:
adding a comment to this part (for some reason can't comment)
`err = k.CreatePost(ctx, vendorID, postID, "", deposit, curator, rewardAccount)` 
Now that I think it , shouldn't the post already exists ? I mean if the post doesn't previously exists on chain a transaction should be submitted before from the creator with `reward_address = curator` 

The conflict is that the post would not contain the real `author` address

Reply:
Ah yes, the post would not contain the real author/creator in this case. Maybe we should make the creator optional? So in this case, the creator would be `nil`, but the `reward_address` would be the curator.

Reply:
Created #76